### ISSUE TITLE
Integer and Float have same DBAPI type

### ISSUE THREAD

**Issue by b-schmeling** [duckdb/duckdb#10495]
### What happens?

When I fetch the DBAPI types from `result.cursor.description`, there is no difference between an Integer and a Double. Because of that, I have to query `information_schema.columns` in order to get detailed type information. 

With other databases, when I use `result.cursor.description` they return their internal type codes. E.g. postgres returns something with the format `(Column(name='a', type_code=23), Column(name='b', typ..._code=701))`

On the [PEP 249 standard](https://peps.python.org/pep-0249/#type-objects), the type codes should be "equal" to one their Type Objects, not identical.
> The type_code must compare equal to one of Type Objects defined below. Type Objects may be equal to more than one type code (e.g. DATETIME could be equal to the type codes for date, time and timestamp columns; see the [Implementation Hints](https://peps.python.org/pep-0249/#implementation-hints) below for details).

I believe that the type codes are meant to cover the full range of internal DB types.

### To Reproduce

```python
import sqlalchemy
from sqlalchemy.sql import text

if __name__ == "__main__":
    engine = sqlalchemy.create_engine(
        "sqlite:///:memory:",
    )
    with engine.begin() as conn:
        conn.execute(text("CREATE TABLE test (a INTEGER, b DOUBLE)"))
        conn.execute(text("INSERT INTO test VALUES (1, 1.1)"))

        result = conn.execute(text("SELECT * FROM test"))

        metadata = result.cursor.description

        print(metadata)

```

Prints: `[('a', 'NUMBER', None, None, None, None, None), ('b', 'NUMBER', None, None, None, None, None)]`

### OS:

linux/amd64

### DuckDB Version:

0.9.2

### DuckDB Client:

Python

### Full Name:

Bethany Schmeling

### Affiliation:

Datachat

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi @b-schmeling, thanks for reporting this. Your reproducer used `"sqlite:///:memory:"`, I assume you mean `"duckdb:///:memory:"`, so I edited your script. We'll take a look.

**Comment by b-schmeling** [comment]
> Your reproducer used "sqlite:///:memory:", I assume you mean "duckdb:///:memory:", so I edited your script

That is correct, thanks for catching that! The result I posted came from using the `"duckdb:///:memory:"` string.

**Comment by tboddyspargo** [comment]
NOTE: We're still struggling with this issue, even on v1.1.1

**Comment by tboddyspargo** [comment]
Having ANY additional information (even precision and scale details) to help distinguish between different numerical types would be a big help here.

**Comment by THX1188** [comment]
Instead of using SQLAlchemys Cursor-Description API, did you try with a Native DuckDb call?
Code generated by X/Grok 3.0 Beta LLM; I know this is not based on the Cursor, but knowing the underlying table
at minimum, you could get the original type;

```python

import sqlalchemy
from sqlalchemy.sql import text

if __name__ == "__main__":
    engine = sqlalchemy.create_engine("duckdb:///:memory:")
    with engine.begin() as conn:
        conn.execute(text("CREATE TABLE test (a INTEGER, b DOUBLE)"))
        conn.execute(text("INSERT INTO test VALUES (1, 1.1)"))

        # Get precise column types
        result = conn.execute(text("DESCRIBE test")).fetchall()
        column_types = {row[0]: row[1] for row in result}
        print(column_types)  # {'a': 'INTEGER', 'b': 'DOUBLE'}

        # Proceed with your logic using column_types
```

**Comment by tboddyspargo** [comment]
> Instead of using SQLAlchemys Cursor-Description API, did you try with a Native DuckDb call?

We haven't tried that, but it seems like it could be a helpful work-around. Thanks for the suggestion!

**Comment by tboddyspargo** [comment]
> Instead of using SQLAlchemys Cursor-Description API,

Just to be clear, though: despite using SQLAlchemy to get the cursor description, it is not inherently part of SQLAlchemy. It's the DBAPI implementation of `duckdb` that provides the cursor and determines the level of detail in the description.

**Comment by Tishj** [comment]
This should be fixed by https://github.com/duckdb/duckdb-python/pull/43
Feel free to drop by there and leave a review 👍
